### ISSUE TITLE
Must be executed under progress indicator: com.intellij.openapi.progress.EmptyProgressIndicator@4c3bb054

### ISSUE THREAD

**Issue by jgcmarins** [jimkyndemeyer/js-graphql-intellij-plugin#115]
Hello, @jimkyndemeyer 
First of all, thanks for this awesome plugin! Great work :)
Well, I am new to WS and got the following error.
## System details:
```
WebStorm 2017.3 EAP
Build #WS-173.3727.2, built on November 14, 2017
WebStorm EAP User
Expiration date: December 14, 2017
JRE: 1.8.0_152-release-1024-b6 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.13.1
```
## Stack trace
```
Must be executed under progress indicator: com.intellij.openapi.progress.EmptyProgressIndicator@4c3bb054. Please see e.g. ProgressManager.runProcess()
java.lang.Throwable: Must be executed under progress indicator: com.intellij.openapi.progress.EmptyProgressIndicator@4c3bb054. Please see e.g. ProgressManager.runProcess()
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:136)
	at com.intellij.openapi.progress.impl.CoreProgressManager.assertUnderProgress(CoreProgressManager.java:734)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processFilesConcurrentlyDespiteWriteActions(PsiSearchHelperImpl.java:343)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processPsiFileRoots(PsiSearchHelperImpl.java:318)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processElementsWithTextInGlobalScope(PsiSearchHelperImpl.java:293)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.bulkProcessElementsWithWord(PsiSearchHelperImpl.java:185)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processElementsWithWord(PsiSearchHelperImpl.java:166)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processElementsWithWord(PsiSearchHelperImpl.java:145)
	at com.intellij.lang.jsgraphql.ide.project.JSGraphQLPsiSearchHelper.resolveFragmentReference(JSGraphQLPsiSearchHelper.java:76)
	at com.intellij.lang.jsgraphql.psi.JSGraphQLNamedTypePsiElement.getReference(JSGraphQLNamedTypePsiElement.java:73)
	at com.intellij.lang.jsgraphql.psi.JSGraphQLNamedTypePsiElement.isErrorInContext(JSGraphQLNamedTypePsiElement.java:146)
	at com.intellij.lang.jsgraphql.ide.annotator.JSGraphQLAnnotator.apply(JSGraphQLAnnotator.java:120)
	at com.intellij.lang.jsgraphql.ide.annotator.JSGraphQLAnnotator.apply(JSGraphQLAnnotator.java:44)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.b(ExternalToolPass.java:191)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.access$700(ExternalToolPass.java:47)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.a(ExternalToolPass.java:165)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1127)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:161)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:337)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:327)
	at com.intellij.util.ui.update.MergingUpdateQueue.lambda$flush$1(MergingUpdateQueue.java:277)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:291)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:246)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:246)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:417)
	at com.intellij.util.Alarm$Request.access$700(Alarm.java:344)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:384)
	at com.intellij.util.Alarm$Request.run(Alarm.java:395)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:242)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


java.lang.Throwable: Must be executed under progress indicator: com.intellij.openapi.progress.EmptyProgressIndicator@5e09cc73. Please see e.g. ProgressManager.runProcess()
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:136)
	at com.intellij.openapi.progress.impl.CoreProgressManager.assertUnderProgress(CoreProgressManager.java:734)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processFilesConcurrentlyDespiteWriteActions(PsiSearchHelperImpl.java:343)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processPsiFileRoots(PsiSearchHelperImpl.java:318)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processElementsWithTextInGlobalScope(PsiSearchHelperImpl.java:293)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.bulkProcessElementsWithWord(PsiSearchHelperImpl.java:185)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processElementsWithWord(PsiSearchHelperImpl.java:166)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processElementsWithWord(PsiSearchHelperImpl.java:145)
	at com.intellij.lang.jsgraphql.ide.project.JSGraphQLPsiSearchHelper.resolveFragmentReference(JSGraphQLPsiSearchHelper.java:76)
	at com.intellij.lang.jsgraphql.psi.JSGraphQLNamedTypePsiElement.getReference(JSGraphQLNamedTypePsiElement.java:73)
	at com.intellij.lang.jsgraphql.psi.JSGraphQLNamedTypePsiElement.isErrorInContext(JSGraphQLNamedTypePsiElement.java:146)
	at com.intellij.lang.jsgraphql.ide.annotator.JSGraphQLAnnotator.apply(JSGraphQLAnnotator.java:120)
	at com.intellij.lang.jsgraphql.ide.annotator.JSGraphQLAnnotator.apply(JSGraphQLAnnotator.java:44)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.b(ExternalToolPass.java:191)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.access$700(ExternalToolPass.java:47)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.a(ExternalToolPass.java:165)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1127)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:161)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:337)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:327)
	at com.intellij.util.ui.update.MergingUpdateQueue.lambda$flush$1(MergingUpdateQueue.java:277)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:291)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:246)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:246)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:417)
	at com.intellij.util.Alarm$Request.access$700(Alarm.java:344)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:384)
	at com.intellij.util.Alarm$Request.run(Alarm.java:395)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:242)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by jimkyndemeyer** [comment]
Hi João.

Thanks for using the plugin.

This was also reported in #109 but was closed since it didn't occur in a later EAP. Are you still seeing this, and do you have specific steps to reproduce that you can share?

/Jim.

**Comment by jgcmarins** [comment]
Hi @jimkyndemeyer, it's good to hear from you.
Unfortunately I don't have specific steps to reproduce. It happens randomly. I guess the main reason is related to the WebStorm version. Both, me and @sibelius, are using an unstable version of WebStorm.
Closing this issues for now. Thanks for you help :)

**Comment by jimkyndemeyer** [comment]
@denofevil Do you have any guidance to offer on this issue?

It's unclear to me whether this is an API change in WebStorm EAP, or whether this plugin needs to change how it resolves references using PsiSearchHelperImpl.processElementsWithWord

Thanks.

**Comment by denofevil** [comment]
@jimkyndemeyer there's a general suggestion that all the reference resolution logic should be moved to `doAnnotate` method as `apply` is supposed to be fast and simple. That move should resolve this problem as well.

**Comment by jimkyndemeyer** [comment]
Thanks for the insight Dennis. Appreciate it.

I'm reopening this issue on that basis.

**Comment by jgcmarins** [comment]
Thanks @jimkyndemeyer and @denofevil

**Comment by jimkyndemeyer** [comment]
Fixed in v1.6.2, pending JetBrains approval.

**Comment by jgcmarins** [comment]
Nice job! Thank you very much.
